tools/ci_build/build.py

```diff
@@ -154,6 +154,7 @@ def parse_arguments():
     parser.add_argument("--use_full_protobuf", action='store_true', help="Use the full protobuf library")
     parser.add_argument("--disable_contrib_ops", action='store_true', help="Disable contrib ops (reduces binary size)")
     parser.add_argument("--skip_onnx_tests", action='store_true', help="Explicitly disable all onnx related tests")
```

Comment:
>skip_onnx_tests [](start = 27, length = 15)

what are onnx tests?   these are like the core onnx, not the onnx runtime  ?  what flag controls if we run onnx rutnime tests?

Reply:
i found it !  it's --test .    if --test is used, all the unit tests run.   can we just go with that and not add a new flag?

---
In reply to: [347567751](https://github.com/microsoft/onnxruntime/pull/2412#discussion_r347567751) [](ancestors = 347567751)

Reply:
It currently goes with `--test` (passing `--test` will call CTest, which runs all tests added with `add_test`, including WinML tests). The `--skip_winml_tests` is a CMake configuration to skip **building**, not running the tests (which consequently also skips running the test, since it's not compiled).

I can build the tests unconditionally, but maybe some users will want to do a quick build without the test suite?

Reply:
So the scenario you want to enable is someone passing "--test --use_wimml" , and they want to build all of winml , run all of the ort tests, but NOT run the winml tests?  

---
In reply to: [347576680](https://github.com/microsoft/onnxruntime/pull/2412#discussion_r347576680) [](ancestors = 347576680)

Reply:
actually ... reading the build.py maybe i don't understand when tests run or don't.   is there a way to not run tests?   is --test always assumed on ?

---
In reply to: [348116463](https://github.com/microsoft/onnxruntime/pull/2412#discussion_r348116463) [](ancestors = 348116463,347576680)

Reply:
You can pass `--build` without `--test`